### DISCUSSION TITLE
Framework recommendations

### DISCUSSION THREAD

**Discussion by grandslammer** [crxjs/chrome-extension-tools#539] [General]
I don't want this to turn into some sort of framework vs framework discussion...

I was just wondering what frameworks you would recommend for:
a) extension development in general
b) what frameworks would you recommend using with CRXJS? Are there any gotchas with any of them etc?

Id would be interested to hear your thoughts!

**Comment by jacksteamdev** [comment]
I think Svelte has great merit as a Chrome Extension framework. It's very light and has first-class support for ShadowDOM, which is important for content scripts.

Documentation for Svelte is coming soon. I'd been holding off b/c there were some HMR hiccups in the content script file writer, but that should be fixed now with CRXJS v2 (in beta now).

Setting up a Svelte-Vite Chrome Extension project is much like the other frameworks. Init using `npm init vite@latest`, then `npm i -D @crxjs/vite-plugin`. 

There's one tweak for Svelte: to use CSS in the ShadowDOM, you need to set the Svelte plugin to handle CSS internally:

```js
// https://vitejs.dev/config/
export default defineConfig({
  plugins: [
    svelte({
      emitCss: false, // don't emit CSS to Vite
    }),
    crx({ manifest }),
  ],
})
```

Instructions on using with Shadow DOM: https://github.com/sveltejs/svelte/pull/5870

**Reply by grandslammer** [reply]
Interesting! I am very much a fan of Svelte as is, so that's great to hear!

Looking forward to reading the CRX JS Svelte documentation when it ready.

**Reply by rzfzr** [reply]
@jacksteamdev are the HMR fixes present on the current beta?
thanks
